index.bs

```diff
@@ -15,19 +15,107 @@ Complain About: accidental-2119 yes, missing-example-ids yes
 Markup Shorthands: markdown yes, css no
 Assume Explicit For: yes
 </pre>
+<pre class="anchors">
+spec: HTTP; urlPrefix: https://httpwg.org/specs/rfc7231.html#
+    type: dfn; text: HTTP 3xx statuses; url: status.3xx
+spec: tracking-dnt; urlPrefix: https://www.w3.org/TR/tracking-dnt/#
+    type: dfn; text: tracking; url: dfn-tracking
+</pre>
 
 <section class="non-normative">
 <h2 id="intro">Introduction</h2>
 
 <em>This section is non-normative.</em>
 
-Blah blah blah.
+Browsers are working to prevent cross-site [=tracking=], which threatens user
+privacy. In addition to third-party cookies and storage, other client-side
+methods exist that enable cross-site tracking. [=Navigational tracking=]
+correlates user identities across sites during navigations between those sites.
+[=Navigational tracking=] uses [=link decoration=] to convey information, but
+not all [=link decoration=] is tracking. This project attempts to distinguish
+tracking from non-tracking navigation and to prevent the tracking without
+damaging similar but benign navigations.
+
 </section>
 
 <h2 id="infra">Infrastructure</h2>
 
 This specification depends on the Infra standard. [[!INFRA]]
 
+# Terminology # {#terminology}
+
+<dfn>Link decoration</dfn> is when the source of a [=hyperlink=] "decorates" its [=URL=]
+with extra information beyond what's necessary to identify the page a user wants
+to navigate to. This information can be placed almost anywhere inside the URL.
```

Comment:
What about cases like https://example.com/login?returnto=item/12345 or https://example.com/auth/callback?token=1234567? By your definition, would these fall in the "necessary to identify the page" category or be acceptable link decoration? I guess I'm wondering if we should extend this definition to include parameters that "enable user-exposed functionality on the page" or something like that.

Reply:
> I feel like those are both (acceptable) link decoration, but the first (https://example.com/login?returnto=item/12345) seems "necessary to identify the page a user wants to navigate to" and so wouldn't be counted as link decoration under my definition. That could indicate a problem with the definition or a problem with my intuition. 🙃 Do you have a sense for which it is?
> 
> The second (https://example.com/auth/callback?token=1234567) is necessary for the page to work, but that can be true for any decoration with the cooperation of the target page, so I want to ensure that the definition can include such cases. The token doesn't identify which page the user wants to see, so I think it counts as "extra information" in the current definition, and so is link decoration. That seems correct to me.
> 
> I've added these to the examples, but I haven't filed github issues for them yet, pending your thoughts.

I agree with you that both are cases of link decoration by your current definition, and it might make sense to keep that definition broader for now.

I wanted to see if we could narrow down the definition of link decoration to the point where any "acceptable" use of parameters is excluded. Thinking about writing browser policy against navigational tracking, it would be great if we had a way to define it without having to carve out cases like auth callback or unsubscribe where it fits the definition but is still somehow acceptable.

But you're right that theoretically any decoration can be constructed to provide some functionality on the site, so I don't have a good solution for this, either. I'm fine with discussing this problem ("can we define acceptable use of navigational tracking?") separately.

Reply:
One thing I'm trying to do with these definitions is to make "link decoration" the sometimes-acceptable thing, and "navigational tracking" be the never-acceptable thing. Your login-related examples do explore some edge cases in even making navigational tracking never-acceptable. I've filed #8 to discuss that further, but I hope to merge these rough definitions before completely figuring that out.